src/apps/chem/test_QCCalculationParametersBase.cc

```diff
@@ -0,0 +1,447 @@
+/*
+ * test_QCCalculationParametersBase.cc
+ *
+ *  Created on: 27 Jun 2019
+ *      Author: fbischoff
+ */
+
+
+
+#include<chem/QCCalculationParametersBase.h>
+#include<madness/mra/mra.h>
+#include<madness/world/parallel_archive.h>
+
+using namespace madness;
+
+class Parameters : public QCCalculationParametersBase {
+public:
+	Parameters() : QCCalculationParametersBase() {
+
+		// initialize with: key, value, comment (optional), allowed values (optional)
+		initialize<double>("econv",1.e-5,"recommended values: 1.e-4 < econv < 1.e-8");
+		initialize<double>("dconv",3.e-4,"recommended values: 1.e-4 < econv < 1.e-8");
+		initialize<int>("maxiter",15);
+		initialize<bool>("localize",true);
+		initialize<std::string>("local","BOYS","localization method",{"boys","pm","canon"});
+		initialize<std::vector<double> >("proto",std::vector<double>{1,2});
+		initialize<std::pair<std::string,double> >("ncf",{"slater",2.0});
+	}
+
+	void read_and_set_derived_values(World& world, std::string inputfile, std::string tag) {
+		read(world,inputfile,tag);
+		set_derived_value("dconv",sqrt(get<double>("econv"))*0.1);
+	}
+
+	// convenience getters
+	double econv() const {return get<double>("econv");}
+	double dconv() const {return get<double>("dconv");}
+	bool localize() const {return get<bool>("localize");}
+	std::string local() const {return get<std::string>("local");}
+	std::pair<std::string,double> ncf() const {return get<std::pair<std::string,double> >("ncf");}
+	int maxiter() const {return get<int>("maxiter");}
+
+
+};
+
+template<typename T>
+void test_same(const T& t1, const T& t2) {
+	if (t1!=t2) {
+		print("t1, t2", t1,t2);
+		std::cout << "++"<< t1 << "++"<< std::endl;
+		std::cout << "++" << t2 << "++" <<std::endl;
+
+		throw std::runtime_error("failure in test");;
+	}
+}
+
+struct inputfile {
+	std::string fname;
+	inputfile(const std::string filename, const std::string lines) {
+		fname=filename;
+		std::ofstream myfile;
+		myfile.open (fname);
+		myfile << lines << std::endl;
+		myfile.close();
+	}
+
+	~inputfile() {
+		remove(fname.c_str());
+	}
+};
+
+bool test_serialize(World& world) {
+
+	std::string inputlines=R"input(mp3 
```

Comment:
need to reformat the whole file (probably other files too)? certainly indentation is not consistent ...